pkg/datapath/linux/routing/info.go

```diff
@@ -46,17 +46,17 @@ type RoutingInfo struct {
 // NewRoutingInfo creates a new RoutingInfo struct, from data that will be
 // parsed and validated. Note, this code assumes IPv4 values because ENI + IPv4
 // is the only supported path currently.
-func NewRoutingInfo(gateway string, cidrs []string, mac string) (*RoutingInfo, error) {
-	return parse(gateway, cidrs, mac)
+func NewRoutingInfo(gateway string, cidrs []string, mac string, masquerade bool) (*RoutingInfo, error) {
```

Comment:
Could you also update the comment of this function to state the intent of `masquerade` and why it's not part of the struct like all the other parameters?